### ISSUE TITLE
twisted.words.service has NameErrors

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#11103]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)| @radix reported|
|-|-|
|Trac ID|trac#1734|
|Type|defect|
|Created|2006-05-22 00:00:51Z|

<details><summary>Searchable metadata</summary>

```
trac-id__1734 1734
type__defect defect
reporter__radix radix
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__words words
keywords__ 
time__1148256051000000 1148256051000000
changetime__1149636030000000 1149636030000000
version__ 
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner|
|-|-|

Fixed in [branches/words-nameerrors-1734](branches/words-nameerrors-1734)

There's no test coverage for the error paths these problems were in, but there's also no "real-world coverage" either - ie, you can't have a permission denied on changing group metadata with the current realm implementation.  When that's possible these codepaths should get some unit tests.

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)|@jml set owner to @jml<br>@jml set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)|@jml set owner to @exarkun<br>@jml set status to `new`|
|-|-|

Looks fine to me.  I think you _should_ have unit tests -- definitely at least one that calls irc_TOPIC directly and passes it an undecodable channel.


Still, OK to merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set status to `closed`|
|-|-|

(In [17093]) Merge words-nameerrors-1734

Author: exarkun
Reviewer: jml
Fixes [#11103](https://github.com/twisted/twisted/issues/11103)

Fix two trivial NameErrors point out by pyflakes in error-handling
codepaths in the IRC support for the multiprotocol chat server.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
